### PR TITLE
make compile with current ros2 HEAD

### PR DESCRIPTION
**Pull request by Karsten1987**
this PR makes it compile with the current ros2 master branch.
Please apologize the whitespace madness, must be a visual studio thing.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ add_custom_target(rclcs ALL
 else()
 add_custom_target(rclcs ALL
 	COMMAND 
-	"msbuild" rclcs.csproj /property:OutputPath=${CMAKE_BINARY_DIR} /property:AssemblyName=rclcs /property:AllowUnsafeBlocks=true /property:Platform=x64 /property:GenerateDocumentation=true /property:DocumentationFile="rclcs.xml"
+	"msbuild" rclcs.csproj /property:OutputPath=${CMAKE_BINARY_DIR} /property:AssemblyName=rclcs /property:Configuration=Release /property:Platform=x64 /property:AllowUnsafeBlocks=true /property:DocumentationFile="rclcs.xml"
```
Comment (firesurfer) [inline CMakeLists.txt]: Is it really necessary to built this as a release build ? Doesn't this make debugging harder?

File: src/Node.cs [hunk 2]
```diff
@@ -298,7 +298,8 @@ public static rcl_node create_native_node (string name)
 		{
 			rcl_node_t node = rcl_get_zero_initialized_node ();
 			rcl_node_options_t default_options = rcl_node_get_default_options ();
-			int ret = rcl_node_init (ref node, name,ref default_options);
+            string ns = "";
```
Comment (firesurfer) [inline src/Node.cs]: Please add call the string namespace and add it as a parameter with "" as default to the method.

File: src/Publisher.cs [hunk 3]
```diff
@@ -84,16 +84,19 @@ public Publisher (Node _Node, string _TopicName, rmw_qos_profile_t _QOS )
 					}
 				}
 			}
-
+            
 			//Now we do the same for the message struct
 			foreach (var item in messageType.GetMethods()) {
-				
+                Console.WriteLine("Found item: " + item.Name);
```
Comment (firesurfer) [inline src/Publisher.cs]: Please remove the debug line

**Comment by firesurfer** [comment]
I added some comments. Could you change these things?

File: CMakeLists.txt (hunk 1 above)
Comment (Karsten1987) [reply CMakeLists.txt]: I revoked that diff, however it was somewhat necessary for me in order to sync the rest of the libs. But maybe also not :)

File: src/Node.cs (hunk 2 above)
Comment (Karsten1987) [reply src/Node.cs]: I hope this is done in the C# correct way, I don't know about any conventions and styles.

**Comment by firesurfer** [comment]
Thanks. I just updated the documentation.

**Pull request merged**